File: benchmarks/oopsla25/subjects/323.py
import sys

def f323(s: str):
  a,x = s[0:2]
  y,b = s[1:]
  assert a == "a"
  assert b == "b"

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f323(my_string)

File: benchmarks/oopsla25/subjects/323.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f323(const char* s) {
    char a = s[0];
    char x = s[1];
    char y = s[1];
    char b = s[2];
    assert(a == 'a');
    assert(b == 'b');
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f323(my_string);
  return 0;
}
